gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/handler/HttpGremlinRequestEncoder.java

```diff
@@ -71,6 +71,7 @@ protected void encode(final ChannelHandlerContext channelHandlerContext, final R
                 gremlin = gremlinStringOrBytecode.toString();
             }
             final byte[] payload = mapper.writeValueAsBytes(new HashMap<String,Object>() {{
+                this.putAll(requestMessage.getArgs());
                 put(Tokens.ARGS_GREMLIN, gremlin);
                 put(Tokens.REQUEST_ID, requestMessage.getRequestId());
```

Comment:
Is this redundant given the above putAll()?

Reply:
request id is not part of Args.

It would be nice to add a test that will confirm that the bindings work correctly

Reply:
The `Tokens.ARGS_GREMLIN` here replaces the `gremlin` stuff in the original request with the processed ones above, so it'd be needed (this was the easier way)